src/main/java/core/controller/car/AddDriverToCarController.java

```diff
@@ -0,0 +1,53 @@
+package core.controller.car;
+
+import core.lib.Injector;
+import core.model.Car;
+import core.model.Driver;
+import core.service.car.CarService;
+import core.service.driver.DriverService;
+import java.io.IOException;
+import java.util.NoSuchElementException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddDriverToCarController extends HttpServlet {
+    private static final Injector injector
+            = Injector.getInstance("core");
+    private final CarService carService = (CarService) injector.getInstance(CarService.class);
+    private final DriverService driverService
+            = (DriverService) injector.getInstance(DriverService.class);
+    
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/car/add_driver_to_car.jsp").forward(req, resp);
+    }
+    
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String carId = req.getParameter("carId");
+        String driverId = req.getParameter("driverId");
+        if (carId.isEmpty() || driverId.isEmpty()) {
+            errorInput(req, resp, "Empty input provided");
+        }
+        try {
+            Driver driver = driverService.get(Long.valueOf(driverId));
+            Car car = carService.get(Long.valueOf(carId));
+            carService.addDriverToCar(driver, car);
+            resp.sendRedirect(req.getContextPath() + "/");
+        } catch (NoSuchElementException e) {
```

Comment:
this is not a good practice to catch RuntimeExceptions
let's throw some custom checked Exception on the service layer instead - or we can just remove this check and assume that ids should be correct (this is even a better option)

Reply:
Ok, then I refactored the method to the second option, as the first one requires changes in the service layer and all the interfaces and that isn't SOLID